pkg/ingester/ingester.go

```diff
@@ -735,19 +784,37 @@ func (i *Ingester) PushWithCleanup(ctx context.Context, pushReq *push.Request) (
 		"failedSamplesCount", stats.failedSamplesCount,
 		"succeededExemplarsCount", stats.succeededExemplarsCount,
 		"failedExemplarsCount", stats.failedExemplarsCount,
+
+		"ephemeralSucceededSamplesCount", ephemeralStats.succeededSamplesCount,
+		"ephemeralFailedSamplesCount", ephemeralStats.failedSamplesCount,
+		"ephemeralSucceededExemplarsCount", ephemeralStats.succeededExemplarsCount,
+		"ephemeralFailedExemplarsCount", ephemeralStats.failedExemplarsCount,
 	)
 
 	startCommit := time.Now()
-	if err := app.Commit(); err != nil {
-		return nil, wrapWithUser(err, userID)
+	if persistentApp != nil {
+		if err := persistentApp.Commit(); err != nil {
+			rollback()
```

Comment:
We'll try to rollback `persistentApp` too (after `Commit()` was called and failed). Is it safe?

Reply:
Documentation for `Commit()` says:

> If Commit returns a non-nil error, it also rolls back all modifications made in the appender so far, as Rollback would do.
> In any case, an Appender must not be used anymore after Commit has been called.

Which tells me that 1) Rollback is automatic in case of errors, 2) we should not use the appender. Let's clear it here. Same for `ephemeralApp` below.